log/lvl.go

```diff
@@ -0,0 +1,377 @@
+
+// Lvlf4 is like Lvl4 but with a format-string
+func Lvlf4(f string, args ...interface{}) {
+	lvlf(4, f, args...)
+}
+
+// Lvlf5 is like Lvl5 but with a format-string
+func Lvlf5(f string, args ...interface{}) {
+	lvlf(5, f, args...)
+}
+
+// LLvl1 *always* prints
+func LLvl1(args ...interface{}) { lvld(-1, args...) }
+
+// LLvl2 *always* prints
+func LLvl2(args ...interface{}) { lvld(-2, args...) }
+
+// LLvl3 *always* prints
+func LLvl3(args ...interface{}) { lvld(-3, args...) }
+
+// LLvl4 *always* prints
+func LLvl4(args ...interface{}) { lvld(-4, args...) }
+
+// LLvl5 *always* prints
+func LLvl5(args ...interface{}) { lvld(-5, args...) }
+
+// LLvlf1 *always* prints
+func LLvlf1(f string, args ...interface{}) { lvlf(-1, f, args...) }
+
+// LLvlf2 *always* prints
+func LLvlf2(f string, args ...interface{}) { lvlf(-2, f, args...) }
+
+// LLvlf3 *always* prints
+func LLvlf3(f string, args ...interface{}) { lvlf(-3, f, args...) }
+
+// LLvlf4 *always* prints
+func LLvlf4(f string, args ...interface{}) { lvlf(-4, f, args...) }
+
+// LLvlf5 *always* prints
+func LLvlf5(f string, args ...interface{}) { lvlf(-5, f, args...) }
+
+// TestOutput sets the DebugVisible to 0 if 'show'
+// is false, else it will set DebugVisible to 'level'
+//
+// Usage: TestOutput( test.Verbose(), 2 )
+func TestOutput(show bool, level int) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+
+	if show {
+		debugVisible = level
+	} else {
+		debugVisible = 0
+	}
+}
+
+// SetDebugVisible set the global debug output level in a go-rountine-safe way
+func SetDebugVisible(lvl int) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	debugVisible = lvl
+}
+
+// DebugVisible returns the actual visible debug-level
+func DebugVisible() int {
+	debugMut.RLock()
+	defer debugMut.RUnlock()
+	return debugVisible
+}
+
+// SetShowTime allows for turning on the flag that adds the current
+// time to the debug-output
+func SetShowTime(show bool) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	showTime = show
+}
+
+// ShowTime returns the current setting for showing the time in the debug
+// output
+func ShowTime() bool {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	return showTime
+}
+
+// SetUseColors can turn off or turn on the use of colors in the debug-output
+func SetUseColors(show bool) {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	useColors = show
+}
+
+// UseColors returns the actual setting of the color-usage in log
+func UseColors() bool {
+	debugMut.Lock()
+	defer debugMut.Unlock()
+	return useColors
+}
+
+// MainTest can be called from TestMain to set up some default
+// variables
+func MainTest(m *testing.M) {
+	flag.Parse()
+	TestOutput(testing.Verbose(), 3)
+	code := m.Run()
+	AfterTest(nil)
+	os.Exit(code)
+}
+
+// ParseEnv looks at the following environment-variables:
+// - DEBUG_LVL - for the actual debug-lvl - default is 1
+// - DEBUG_TIME - whether to show the timestamp - default is false
+// - DEBUG_COLOR - whether to color the output - default is true
+func ParseEnv() {
```

Comment:
I prefer having this exported, so that the user has control whether he wants this to be automated or not. But if you and @Liamsi think it's OK to put that in `init`, I'll put it there.